test/optim/test_optimize.py

```diff
@@ -1919,3 +1954,7 @@ def my_gen():
         )
         ic_generator = opt_inputs.get_ic_generator()
         self.assertIs(ic_generator, my_gen)
+
+if __name__ == "__main__":
+    import pytest
+    pytest.main([__file__])
```

Comment:
```suggestion

```

Reply:
yep was planning to clean this out before changing the status from draft. It makes it more convenient to run with debugger whilst trying to work out the test failures induced by the change. Will rebase the commits at the end into logical parts.